kotest-common/src/jvmMain/kotlin/io/kotest/common/nonConstantResults.kt

```diff
@@ -0,0 +1,9 @@
+package io.kotest.common
+
+/** returns `true` while preventing the compiler from optimizing it away. */
+@KotestInternal
+fun nonConstantTrue() = System.currentTimeMillis() > 0L
```

Comment:
the name `nonConstantTrue` does not seem intuitive to me. how about `alwaysTrueNotOptimizedAway`?

Reply:
`NotOptimizedAway` is not guaranteed by the code. So while that's the intention, it is also a bit misleading. In any case, this stuff is internal only and I my view probably not worth another iteration.